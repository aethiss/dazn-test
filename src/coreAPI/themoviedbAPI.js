import {
  BASE_URL,
  APIKEY,
  TRENDING_URL,
  MOVIE_SEARCH,
  MOVIE_URL } from '../constants/movieDbUrls'

export const searchMovie = (getUrl) => {
  return new Promise((resolve, reject) => {
    fetch(getUrl, {
      method: 'GET',
      headers: {
        'Accept': 'application/json, text/plain, */*',
        'Content-Type': 'application/json'
      },
    })
    .then((resp) => resp.json())
      .then((data) => {
        // console.log('DATA ===> ', data)
        resolve(data)
      })
      .catch((err) => {
        // console.log('error ::::: ', err)
        reject(err)
      })
  })
}

export const generateRequestUrl = (type = 'trending', value = '') => {
  switch(type) {
    case 'search':
      return `${BASE_URL}${MOVIE_SEARCH}?api_key=${APIKEY}&query=${value}`
      break;
    case 'details':
      // code block movie_id
      return `${BASE_URL}${MOVIE_URL}/${value}?api_key=${APIKEY}`
      break;
    default:
    return `${BASE_URL}${TRENDING_URL}?api_key=${APIKEY}`
  }
}
