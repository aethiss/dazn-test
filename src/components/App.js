import React, { Component } from 'react'

import Home from './container/home/Home'
import Header from "./common/header/Header"

import { searchMovie, generateRequestUrl } from '../coreAPI/themoviedbAPI'

export default class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      stringSearch: '',
      searchResult: []
    }
  }

  updateSearch = (ele) => {
    const strFromSearch = ele.target.value
    if (ele.keyCode === 13 && strFromSearch.length >= 3) {

      searchMovie(generateRequestUrl('search', strFromSearch)).then((res) => {
        if (res.results) {
          this.setState({
            stringSearch: strFromSearch,
            searchResult: res.results || []
          })
        }
      })
    }
  }

  render() {
    return (
      <div>
        <Header updateSearch={this.updateSearch} />
        <main>
          <Home {...this.state}/>
        </main>
      </div>
    )
  }
}
