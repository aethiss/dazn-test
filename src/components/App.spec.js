import React from 'react'
import { shallow, mount } from "enzyme"

import App from './App'

import * as api from '../coreAPI/themoviedbAPI'
const mySpy = jest.spyOn(api, 'searchMovie').mockImplementation(() => {
  return Promise.resolve({ results: ['a', 'b'] })
})

const mockProps = {
  stringSearch: '',
  searchResult: []
}

describe('@App', () => {
  describe('@render', () => {
    it('Default render', () => {
      const component = shallow(<App {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
    it('should render home component', () => {
      const wrapper = mount(
        <App {...mockProps} />
      )
      expect(wrapper.find('.HomeContainer').length).toBe(1)
    })
  })
})
