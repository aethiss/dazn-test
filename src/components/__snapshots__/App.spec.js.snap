// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@App @render Default render 1`] = `
<div>
  <Header
    updateSearch={[Function]}
  />
  <main>
    <Home
      searchResult={Array []}
      stringSearch=""
    />
  </main>
</div>
`;
